src/analysisd/decoders/security_configuration_assessment.c

```diff
@@ -841,20 +841,21 @@ static void HandleCheckEvent(Eventinfo *lf,int *socket,cJSON *event) {
                         char *value = NULL;
                         int free_value = 0;
 
-                        if(!comp->valuestring){
-                            if(comp->valueint) {
-                                os_calloc(OS_SIZE_1024, sizeof(char), value);
-                                sprintf(value, "%d", comp->valueint);
-                                free_value = 1;
-                            } else if(comp->valuedouble) {
-                                os_calloc(OS_SIZE_1024, sizeof(char), value);
-                                sprintf(value, "%lf", comp->valuedouble);
-                                free_value = 1;
+                        if(cJSON_IsString(comp) && (comp->valuestring != NULL)) {
+                            value = comp->valuestring;
+                        } else if (cJSON_IsNumber(comp)) {
+                            os_calloc(OS_SIZE_1024, sizeof(char), value);
+                        
+                            if(comp->valuedouble == (double)comp->valueint) {
+                                snprintf(value, sizeof(value), "%d", comp->valueint);
+                            } else {
+                                snprintf(value, sizeof(value), "%lf", comp->valuedouble);
```

Comment:
In this case, sizeof(value) returns the size of the pointer (8 bytes), not the capacity of the buffer. 
This is because value is not a buffer but a pointer char *value = NULL;

We should use OS_SIZE_1024 (allocated memory):
```suggestion
                                snprintf(value, OS_SIZE_1024, "%d", comp->valueint);
                            } else {
                                snprintf(value, OS_SIZE_1024, "%lf", comp->valuedouble);
```